src/main/java/com/coveo/pushapiclient/StreamServiceInternal.java

```diff
@@ -8,52 +9,120 @@
 
 /** For internal use only. Made to easily test the service without having to use PowerMock */
 class StreamServiceInternal {
-  private Logger logger;
+  private final Logger logger;
   private final StreamEnabledSource source;
   private final PlatformClient platformClient;
   private String streamId;
-  private DocumentUploadQueue queue;
+  private FileContainer fileContainer;
+  private final DocumentUploadQueue queue;
+  private final StreamOperationType operationType;
 
   public StreamServiceInternal(
       StreamEnabledSource source,
       DocumentUploadQueue queue,
       PlatformClient platformClient,
+      StreamOperationType operationType,
       Logger logger) {
     this.source = source;
     this.queue = queue;
     this.platformClient = platformClient;
+    this.operationType = operationType;
     this.logger = logger;
   }
 
   public String add(DocumentBuilder document) throws IOException, InterruptedException {
+    if (StreamOperationType.INCREMENTAL.equals(getOperationType())) {
+      throw new UnsupportedOperationException(
+          "The incremental stream operation is not supported for the rebuild add method");
+    }
     if (this.streamId == null) {
-      this.streamId = this.getStreamId();
+      this.streamId = this.openStream();
     }
     queue.add(document);
     return this.streamId;
   }
 
+  public FileContainer addOrUpdate(DocumentBuilder document)
+      throws IOException, InterruptedException {
+    if (StreamOperationType.REBUILD.equals(getOperationType())) {
```

Comment:
Perhaps it's more resilient if the logic was inverted, for when other operation types are added in the future like "partial item updates"? e.g.

```
if (!StreamOperationType.INCREMENTAL.equals(getOperationType())) {...}
```